File: src/par2/par2/compare_speed.py
gf = [i for i in range(16)]
gfi = [i for i in range(16)]

for j in range(100000):
    if True:
        for i in range(16):
            c = gf[i] + gf[i]
            if c >= 16:
                c -= 16
            y = gfi[c]
    else:
        for i in range(16):
            c = i * i
            if c >= 16:
                c &= 15

File: src/par2/par2/compare_speed.c
#include <stdio.h>

#define N 16

int main(void)
{
#if 0
    int i, j, n;
    int times = 100000;
    int total = 0;

    for(times=0;times<1000000;times++){
        for(j=0;j<N;j++){
            for(i=0;i<N;i++){
                n = i * j;
                n &= (N - 1);
                total += n;
            }
        }
    }
    printf("total = %d\n", total);
#else
    int i, j, n;
    int times = 100000;
    int gf[N], gfi[N];
    int total = 0;

    for(times=0;times<1000000;times++){
        for(j=0;j<N;j++){
            for(i=0;i<N;i++){
                n = gf[i] + gf[j];
                total += n;
            }
        }
    }
    printf("total = %d\n", total);
#endif

    return 0;
}
